build_wheels.py

```diff
@@ -172,28 +177,239 @@ def assemble_requirements(idf_branches: List[str], idf_constraints: List[str]) -
+    for requirement in requirements:
+        # --only-binary requirement wheel build
+        if non_classic_requirement and non_classic_requirement[0].replace('--only-binary ', '') in requirement.name:
+            only_bin = non_classic_requirement[0].replace('--only-binary', '').strip()
+            out = subprocess.run(
+                ['python', '-m', 'pip', 'wheel', f'{requirement}',
+                 '--find-links', f'{dir}', '--wheel-dir', f'{dir}', '--only-binary', f'{only_bin}'],
+                 stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                )
+            print(out.stdout, out.stderr)
+            non_classic_requirement.remove(non_classic_requirement[0])
+            continue
+
+        # requirement wheel build
+        out = subprocess.run(
+            ['python', '-m', 'pip', 'wheel', f'{requirement}',
+             '--find-links', f'{dir}', '--wheel-dir', f'{dir}'],
+             stdout=subprocess.PIPE, stderr=subprocess.PIPE
+             )
+
+        print(out.stdout.decode('utf-8'))
+        if out.stderr:
+            print(Fore.RED + out.stderr.decode('utf-8') + Style.RESET_ALL)
+
+        if out.returncode != 0:
+            failed_wheels += 1
+        else:
+            succeeded_wheels += 1
 
 
 def main():
-    """Builds Python wheels for IDF dependencies"""
+    """Builds Python wheels for ESP-IDF dependencies for master and release branches
+    grater or equal to specified"""
+
     idf_repo_branches = fetch_idf_branches()
     idf_branches = get_used_idf_branches(idf_repo_branches)
     print(f'ESP-IDF branches to be downloaded requirements for:\n{idf_branches}\n')
 
     idf_constraints = get_constraints_versions(idf_branches)
     print(f'ESP-IDF constrains files versions to be downloaded requirements for:\n{idf_constraints}\n')
 
-    requirements_set = assemble_requirements(idf_branches, idf_constraints)
+    requirements = assemble_requirements(idf_branches, idf_constraints)
+
+    exclude_list = yaml_to_requirement('exclude_list.yaml', exclude=True)
+    excluded_requirements = exclude_from_requirements(requirements, exclude_list)
+
+    include_list = yaml_to_requirement('include_list.yaml')
+    print(Fore.BLUE + '---------- ADDITIONAL REQUIREMENTS ----------' + Style.RESET_ALL)
+    for req in include_list:
+        print(req)
+    print(Fore.BLUE + '---------- END OF ADDITIONAL REQUIREMENTS ----------' + Style.RESET_ALL)
+
+    print(Fore.BLUE + '---------- BUILD ADDITIONAL WHEELS ----------' + Style.RESET_ALL)
+    build_wheels(include_list)
+
+    print(Fore.BLUE + '---------- BUILD WHEELS ----------' + Style.RESET_ALL)
+    build_wheels(excluded_requirements)
+
+    print(Fore.BLUE + '---------- STATISTICS ----------')
+    print(Fore.GREEN + f'Succeeded {succeeded_wheels} wheels')
+    print(Fore.RED + f'Failed {failed_wheels} wheels' + Style.RESET_ALL)
 
-    print_requirements(requirements_set)
+    assert failed_wheels == 0
```

Comment:
(optional) You could use this to generate an exit code. It would be more user-friendly then throwing an AssertionError.

Reply:
To proper handle exit code from the script ... FYI https://github.com/espressif/conventional-precommit-linter/blob/master/conventional_precommit_linter/hook.py#L223-L224